labgrid/target.py

```diff
@@ -126,10 +126,30 @@ def bind_driver(self, client):
 
         # locate suppliers
         suppliers = []
-        for name, cls in client.bindings.items():
-            supplier = self.get(cls)
-            setattr(client, name, supplier)
-            suppliers.append(supplier)
+        for name, requirements in client.bindings.items():
+            # use sets even for a single requirement
+            if not isinstance(requirements, set):
+                requirements = {requirements}
+            errors = []
+            suppliers = []
+            for requirement in requirements:
+                try:
+                    suppliers.append(self.get(requirement))
+                except NoSupplierFoundError as e:
+                    errors.append(e)
+            if not suppliers:
+                if len(errors) == 1:
+                    raise errors[0]
```

Comment:
If we only have a single error, I wanted to pass that to the caller. This way, the more specific exception (NoDriverFoundError/NoResourceFoundError) class is still available.